src/components/StatusWidget.tsx

```diff
@@ -41,39 +44,51 @@ export class StatusWidget extends ReactWidget {
 
   render(): JSX.Element {
     return (
-      <UseSignal
-        signal={this._model.credentialsRequiredChanged}
-        initialArgs={false}
-      >
-        {(_, needsCredentials) => (
-          <Badge
-            className={badgeClass}
-            variant="dot"
-            invisible={!needsCredentials}
-            data-test-id="git-credential-badge"
-          >
-            <ActionButton
-              className={classes(
-                toolbarButtonClass,
-                this._status !== 'idle'
-                  ? statusAnimatedIconClass
-                  : statusIconClass
-              )}
-              icon={gitIcon}
-              onClick={
-                needsCredentials
-                  ? async () => this._showGitOperationDialog()
-                  : undefined
-              }
-              title={
-                needsCredentials
-                  ? `Git: ${this._trans.__('credentials required')}`
-                  : `Git: ${this._trans.__(this._status)}`
-              }
-            />
-          </Badge>
-        )}
-      </UseSignal>
+      <>
+        <UseSignal
+          signal={this._model.credentialsRequiredChanged}
+          initialArgs={false}
+        >
+          {(_, needsCredentials) => (
+            <Badge
+              className={badgeClass}
+              variant="dot"
+              invisible={!needsCredentials}
+              data-test-id="git-credential-badge"
+            >
+              <ActionButton
+                className={classes(
+                  toolbarButtonClass,
+                  this._status !== 'idle'
+                    ? statusAnimatedIconClass
+                    : statusIconClass
+                )}
+                icon={gitIcon}
+                onClick={
+                  needsCredentials
+                    ? async () => this._showGitOperationDialog()
+                    : undefined
+                }
+                title={
+                  needsCredentials
+                    ? `Git: ${this._trans.__('credentials required')}`
+                    : `Git: ${this._trans.__(this._status)}`
+                }
+              />
+            </Badge>
+          )}
+        </UseSignal>
+
+        <UseSignal signal={this._model.headChanged}>
+          {() =>
+            this._model.currentBranch && (
+              <span className={currentBranchNameClass}>
+                {this._model.currentBranch.name}*
```

Comment:
Let's remove the star for now.

```suggestion
                {this._model.currentBranch.name}
```